### PR TITLE
Added a example for seed-based correlation and fixed a typo in add_marker

### PR DESCRIPTION
**Pull request by fliem**
A thing that's not too pretty yet, is that add_marker takes marker_size in points. Is there a way to give this in mm (as the MaskerSphere is also defined in mm) 
See #1019

### CODE REVIEW

File: examples/03_connectivity/plot_seed_based_correlation.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
+"""
+Producing single subject maps of seed-based correlation
+========================================================
+
+This example shows how to produce seed-based correlation maps for a single subject based on resting-state fMRI scans.
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_seed_based_correlation.py]: Your lines are definitely long than 79 chars :)

File: examples/03_connectivity/plot_seed_based_correlation.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+#############################
+# Time series extraction
+# ----------------------
+#
+# We are going to extract signals from the functional time series in two steps. First we will extract the mean signal within
+# the **seed region of interest**. Second, we will extract the **brain-wide voxel-wise time series**.
+#
+# We will be working with one seed sphere in the Posterior Cingulate Cortex, considered part of the Default Mode Network.
+dmn_coords = [(0, -52, 18)]
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_seed_based_correlation.py]: Could you call this pcc_coords please

File: examples/03_connectivity/plot_seed_based_correlation.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    memory='nilearn_cache', memory_level=1, verbose=0)
+
+##########################################################################
+# Then we extract the brain-wide voxel-wise time series while regressing out the confounds as before
+brain_time_series = brain_masker.fit_transform(func_filename, confounds=[confound_filename])
+
+
+#############################
+# Inspecting the extracted time series
+# ----------------------
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_seed_based_correlation.py]: Can you make the line below the title as long as the text (to be valid restructured text).

File: examples/03_connectivity/plot_seed_based_correlation.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+################################################
+# Plotting the seed-based correlation map
+# ----------------------------------------------
+# We can also plot this image and perform thresholding to only show values more extreme than +/- 0.3.
+# Furthermore, we can display the location of the seed with a sphere and set the cross to the center
+# of the seed region of interest.
+from nilearn import plotting
+
+display = plotting.plot_stat_map(seed_based_correlation_img, threshold=0.3, cut_coords=dmn_coords[0])
+display.add_markers(marker_coords=np.array(dmn_coords), marker_color='g', marker_size=300)
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_seed_based_correlation.py]: Do you need the "np.array" here? If so, we need to fix add_markers.

Comment (fliem) [inline examples/03_connectivity/plot_seed_based_correlation.py]: It fails without np.array

  File "<string>", line 6, in <module>
  File "/Users/franzliem/Dropbox/Workspace/nilearn/nilearn/plotting/displays.py", line 740, in add_markers
    marker_coords, direction, return_direction=True)
  File "/Users/franzliem/Dropbox/Workspace/nilearn/nilearn/plotting/displays.py", line 209, in _coords_3d_to_2d
    return coords_3d[:, dimensions], coords_3d[:, index]
TypeError: list indices must be integers, not tuple

File: examples/03_connectivity/plot_seed_based_correlation.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
+"""
+Producing single subject maps of seed-based correlation
+========================================================
+
+This example shows how to produce seed-based correlation maps for a single subject based on resting-state fMRI scans.
+These maps depict the temporal correlation of a **seed region** with the **rest of the brain**.
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_seed_based_correlation.py]: I would add here: that "This example is an advanced one that requires manipulating the data with numpy.

Note the difference between images, that lie in brain space space, and the numpy array, corresponding to the data inside the mask".

File: examples/03_connectivity/plot_seed_based_correlation.py [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+# Time series extraction
+# ----------------------
+#
+# We are going to extract signals from the functional time series in two
+# steps. First we will extract the mean signal within the **seed region of
+# interest**. Second, we will extract the **brain-wide voxel-wise time series**.
+#
+# We will be working with one seed sphere in the Posterior Cingulate Cortex,
+# considered part of the Default Mode Network.
+dmn_coords = [(0, -52, 18)]
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_seed_based_correlation.py]: Could you name this variable "pcc_coords", as it's really the pcc, not the DMN.

File: examples/03_connectivity/plot_seed_based_correlation.py [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+# out the confounds as before
+brain_time_series = brain_masker.fit_transform(func_filename,
+                                               confounds=[confound_filename])
+
+
+##########################################################################
+# Inspecting the extracted time series
+# ------------------------------------
+
+##########################################################################
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_seed_based_correlation.py]: Why the extra split?

Comment (fliem) [inline examples/03_connectivity/plot_seed_based_correlation.py]: As not to make the chapter too long. Should I merge the two?

Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_seed_based_correlation.py]: > Should I merge the two?

Please!

File: examples/03_connectivity/plot_seed_based_correlation.py [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+# Then we extract the mean time series within the seed region while
+# regressing out the confounds that
+# can be found in the dataset's csv file
+seed_time_series = seed_masker.fit_transform(func_filename,
+                                             confounds=[confound_filename])
+
+##########################################################################
+# Next, we can proceed similarly for the **brain-wide voxel-wise time
+# series**, using :class:`nilearn.in_data.NiftiMasker` with the same input
+#  arguments as in the seed_masker in addition to smoothing with a 6 mm kernel
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_seed_based_correlation.py]: Darn, there is an indentation error in the line above.

File: examples/03_connectivity/plot_seed_based_correlation.py [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+                          seed_time_series.shape[0]
+
+################################################
+# The resulting array will contain a value representing the correlation
+# values between the signal in the **seed region** of interest and **each
+# voxel's signal**, and will be of shape (n_voxels, 1). The correlation
+# values can potentially range between -1 and 1.
+print("seed-based correlation shape: (%s, %s)" % seed_based_correlations.shape)
+print("seed-based correlation: min = %.3f; max = %.3f" % (
+seed_based_correlations.min(), seed_based_correlations.max()))
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_seed_based_correlation.py]: Code formatting guidelines say that you should indent the "seed_based_correlation" to match the right after the opening parenthesis.

Comment (fliem) [inline examples/03_connectivity/plot_seed_based_correlation.py]: Like this?

Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_seed_based_correlation.py]: Yes!

File: examples/03_connectivity/plot_seed_based_correlation.py [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+##########################################################################
+# Then we extract the mean time series within the seed region while
+# regressing out the confounds that
+# can be found in the dataset's csv file
+seed_time_series = seed_masker.fit_transform(func_filename,
+                                             confounds=[confound_filename])
+
+##########################################################################
+# Next, we can proceed similarly for the **brain-wide voxel-wise time
+# series**, using :class:`nilearn.in_data.NiftiMasker` with the same input
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_seed_based_correlation.py]: It's actually "input_data", rather than "in_data"

Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_seed_based_correlation.py]: I'll fix that and merge. Thanks!

**Comment by GaelVaroquaux** [comment]
I merged. Thank you!

**Pull request closed without merging**